fs-storage/src/folder_storage.rs

```diff
@@ -195,11 +206,73 @@ where
         Ok(())
     }
 
-    /// Compare the timestamp of the storage file
-    /// with the timestamp of the in-memory storage and the last written
+    /// Compare the timestamp of the storage files
+    /// with the timestamps of the in-memory storage and the last written
     /// to time to determine if either of the two requires syncing.
     fn sync_status(&self) -> Result<SyncStatus> {
-        unimplemented!()
+        let mut ram_newer = false;
+        let mut disk_newer = false;
+
+        for (key, ram_timestamp) in &self.ram_timestamps {
+            let file_path = self.path.join(format!("{}.bin", key));
+
+            if let Ok(metadata) = fs::metadata(&file_path) {
+                if let Ok(disk_timestamp) = metadata.modified() {
+                    match ram_timestamp.cmp(&disk_timestamp) {
+                        std::cmp::Ordering::Greater => ram_newer = true,
+                        std::cmp::Ordering::Less => disk_newer = true,
+                        std::cmp::Ordering::Equal => {}
+                    }
+                } else {
+                    // If we can't read the disk timestamp, assume RAM is newer
+                    ram_newer = true;
```

Comment:
Adding some `log::debug` statements where the a variable changes and which file caused it to change will help in debugging later in case tests/app fails.